src/include/distributed/version_compat.h

```diff
@@ -24,6 +24,45 @@
 #include "optimizer/optimizer.h"
 #endif
 
+#if (PG_VERSION_NUM >= PG_VERSION_13)
+#include "tcop/tcopprot.h"
+#endif
+
+#if PG_VERSION_NUM >= PG_VERSION_13
+#define heap_open(r, l) table_open(r, l)
+#define heap_openrv(r, l) table_openrv(r, l)
+#define heap_openrv_extended(r, l, m) table_openrv_extended(r, l, m)
+#define heap_close(r, l) table_close(r, l)
+#define lnext_compat(l, r) lnext(l, r)
+#define list_delete_cell_compat(l, c, p) list_delete_cell(l, c)
+#define pg_plan_query_compat(p, q, c, b) pg_plan_query(p, q, c, b)
+#define planner_compat(p, c, b) planner(p, NULL, c, b)
+#define standard_planner_compat(a, c, d) standard_planner(a, NULL, c, d)
+#define PortalDefineQuerySelectCompat(a, b, c, d, e) PortalDefineQuery(a, b, c, \
+																	   CMDTAG_SELECT, d, \
+																	   e)
+#define getOwnedSequencesCompat(a, b) getOwnedSequences(a)
+#define ExplainOnePlanCompat(a, b, c, d, e, f, g) ExplainOnePlan(a, b, c, d, e, f, g, \
+																 NULL)
+#define varoattno varattnosyn
+#define varnoold varnosyn
```

Comment:
this seems a bit sketchy, this renames arbitrary variables

Reply:
Right, I guess we can use Compat functions to set, get them in a nicer way.

Reply:
you went for a solution where we still do variable name rewriting I believe, instead of set marco's. Did you try that and did it fail?

Would be great to explain why we didn't go that route in the end.

Reply:
Thinking about this a bit more, using what postgres has in our codebase might be better and that way when we remove pg12 we will only need to remove the macro.

If we use a macro like `Set.....Compat` then we will need to change the code when we are dropping 12 support.

To me, in terms of development or other reasons I don't see a downside of having the up-to-date variable name in the codebase and doing the mapping in the preprocessor because we probably don't need to know the old names while coding.

One downside as Marco mentioned is that this will rename arbitrary variables. I am not sure how big of a downside that is with these variables.

Considering the cons/pros, it seems a bit neutral to me, what do you think @marcocitus?

Reply:
I'm mainly thinking about cases where we copy postgres code and renaming changes the semantics, which would be a debugging nightmare. I suppose C might be just strict enough to end up with some compilation error/warning most of the time and these names are quite exotic, but it's not a great precedent to set.

I'm fine with merging as is, but I'll continue to push back on clever hacks in the future :).

Reply:
Thanks for keeping an eye out for clever hacks! Please keep doing.

I had some doubts here as well, but also agree with the complexities of maintenance were we to create compatibility macros for setting and reading these values.

The good thing from copying code (as long as it comes from pg13) is that is now will also compile for pg 11 and 12.
If we will ever run into strange bugs due to this we can rewrite. The blame of this change will hopefully trigger the finder to find this conversation and tell us we were out of our mind when we merged this.